src/popsift/common/assist.h

```diff
@@ -93,6 +106,34 @@ static inline unsigned int microhash( int val )
                        ^ ( ( val & ( 0xf << 28 ) ) >> 28 ) );
     return ret;
 }
-#define DERR std::cerr << std::hex << popsift::microhash(pthread_self()) << std::dec << "    "
+#define DERR std::cerr << std::hex << popsift::microhash(getCurrentThreadId()) << std::dec << "    "
+
+
+__host__
+static size_t getPageSize()
+{
+#ifdef _WIN32
+    SYSTEM_INFO si;
+    GetSystemInfo(&si);
+    return si.dwPageSize;
+#else
+    return sysconf(_SC_PAGESIZE);
+#endif
+}
+
+static void* memalign(size_t alignment, size_t size)
+{
+#ifdef _WIN32
+    return _aligned_malloc(size, alignment);
```

Comment:
This is not a sufficient fix. Whereas memory obtained with `posix_memalign` can be freed with `free`, memory obtained with `_aligned_malloc` must be freed with `_aligned_free`. The comment in the example explicitly states that using free is illegal: https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/aligned-malloc

Also, you need to include `<malloc.h>` to use aligned functions.

Reply:
Actually, since you're using alignment to page size, it'd be better to use anonymous `mmap` on unix or `VirtualAlloc` on Windows. When freed (`munmap` / `VirtualFree`) it's guaranteed that the memory is immediately returned to the OS instead to the process's heap. This in turn can be encapsulated into a single class, say `AlignedBuffer`.